File: src/com/mojang/escape/level/block/LadderBlock.java
package com.mojang.escape.level.block;

import com.mojang.escape.*;
import com.mojang.escape.entities.*;
import com.mojang.escape.gui.BasicSprite;
import com.mojang.escape.gui.Sprite;

public class LadderBlock extends Block {
	private static final int LADDER_COLOR = 0xDB8E53;
	public boolean wait;

	public LadderBlock(boolean down) {
		if (down) {
			setFloorTex(1);
			addSprite(new BasicSprite(0, 0, 0, 8 + 3, Art.INSTANCE.getCol(LADDER_COLOR)));
		} else {
			setCeilTex(1);
			addSprite(new BasicSprite(0, 0, 0, 8 + 4, Art.INSTANCE.getCol(LADDER_COLOR)));
		}
	}

	public void removeEntity(Entity entity) {
		super.removeEntity(entity);
		if (entity instanceof Player) {
			wait = false;
		}
	}

	public void addEntity(Entity entity) {
		super.addEntity(entity);

		if (!wait && entity instanceof Player) {
			getLevel().switchLevel(getId());
			Sound.Companion.getLadder().play();
		}
	}
}

File: src/com/mojang/escape/level/block/LadderBlock.kt
package com.mojang.escape.level.block

import com.mojang.escape.Art
import com.mojang.escape.Sound
import com.mojang.escape.entities.Entity
import com.mojang.escape.entities.Player
import com.mojang.escape.gui.BasicSprite

class LadderBlock(down: Boolean): Block() {
    companion object {
        private const val LADDER_COLOR = 0xDB8E53
    }
    var wait: Boolean = false

    init {
        if (down) {
            floorTex = 1
            addSprite(BasicSprite(0.0, 0.0, 0.0, 8 + 3, Art.getCol(LADDER_COLOR)))
        } else {
            ceilTex = 1
            addSprite(BasicSprite(0.0, 0.0, 0.0, 8 + 4, Art.getCol(LADDER_COLOR)))
        }
    }

    override fun removeEntity(entity: Entity) {
        super.removeEntity(entity)
        if (entity is Player) {
            wait = false
        }
    }

    override fun addEntity(entity: Entity) {
        super.addEntity(entity)

        if (!wait && entity is Player) {
            level?.switchLevel(id)
            Sound.ladder.play()
        }
    }
}